File: backend/app/api/routes/checkin.py
from fastapi import APIRouter, BackgroundTasks

from app.api.deps import DbDep, OrganizadorUser
from app.schemas.checkin import CheckinRequest, CheckinResponse
from app.service.ingresso_service import validar_checkin

router = APIRouter(tags=["Check-in"])


@router.post("/checkin", response_model=CheckinResponse)
async def checkin_ingresso(
    data: CheckinRequest,
    db: DbDep,
    organizador: OrganizadorUser,
    background_tasks: BackgroundTasks,
):
    return await validar_checkin(
        db,
        qr_code_hash=data.qr_code_hash,
        usuario=organizador,
        background_tasks=background_tasks,
    )

File: frontend/src/api/checkin.ts
// Check-in via QR Code — UC04. Apenas organizador autenticado.

import { api } from "./client";

export type CheckinRequest = {
  qr_code_hash: string;
};

export type CheckinResponse = {
  checkin_id: string;
  ingresso_id: string;
  realizado_em: string;
};

export const realizarCheckin = async (
  payload: CheckinRequest,
): Promise<CheckinResponse> => {
  const { data } = await api.post<CheckinResponse>("/checkin", payload);
  return data;
};
